### PR TITLE
[bacon recoup] Ports a bunch of init & subsystem loggings/fixes

### PR DESCRIPTION
**Pull request by Tsar-Salat**
## About The Pull Request

recoup of #7285 

I think with the additional changes made in this pr as well as the changes in the last 2 years, this should go over better

Its just a bunch of slapped on changes, so Ill explain each pr:

1. Init logging is now in its own file.
2. Background ss no longer get allocated the entirety of the tick
3. Initialization now occurs in stages: Early, Main, and Max.
4. SSinit time measurements now are precise to 2 decimals
5. ss inits now explictly have failure or success returns 

Ports:
- https://github.com/tgstation/tgstation/pull/62545
- https://github.com/tgstation/tgstation/pull/64498
- https://github.com/tgstation/tgstation/pull/66473
- https://github.com/tgstation/tgstation/pull/69533
- https://github.com/tgstation/tgstation/pull/69775

## Why It's Good For The Game

The frequent MC freezes are becoming a headache.

These SS and MC improvements should add some speed. Logging should show errors if any subsystems have bad inits

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

[0a29982](https://github.com/BeeStation/BeeStation-Hornet/pull/10933/commits/0a29982d23a9bf4b3abd963a8a11081bbde92e41)

</details>

## Changelog
:cl: rkz, mothblocks, Tastyfish, MrStonedOne, LemonInTheDark
server: ss init stages
server: ss init logging
server: ss init time measurement messages now precise to 2 decimals
/:cl:

### CODE REVIEW

**Comment by Tsar-Salat** [comment]
Needless to say, this absolutely needs a TM label

File: code/controllers/master.dm [hunk 1]
```diff
@@ -220,34 +223,50 @@ GLOBAL_REAL(Master, /datum/controller/master) = new
...
+	// Sort subsystems by display setting for easy access.
+	sortTim(subsystems, /proc/cmp_subsystem_display)
 	var/start_timeofday = REALTIMEOFDAY
-	// Initialize subsystems.
-	current_ticklimit = CONFIG_GET(number/tick_limit_mc_init)
-	for (var/datum/controller/subsystem/SS in subsystems)
-		if (SS.flags & SS_NO_INIT || SS.initialized) //Don't init SSs with the correspondig flag or if they already are initialzized
-			continue
-		current_initializing_subsystem = SS
-		log_world("Initializing [SS.name] subsystem.")
```
Comment (EvilDragonfiend) [inline code/controllers/master.dm:235]: It's our own line. Revive this.

File: code/controllers/master.dm [hunk 2]
```diff
@@ -259,8 +278,79 @@ GLOBAL_REAL(Master, /datum/controller/master) = new
...
+	var/static/list/valid_results = list(
+		SS_INIT_FAILURE,
+		SS_INIT_NONE,
+		SS_INIT_SUCCESS,
+		SS_INIT_NO_NEED,
+	)
+
+	if (subsystem.flags & SS_NO_INIT || subsystem.initialized) //Don't init SSs with the corresponding flag or if they already are initialized
+		return
+
```
Comment (EvilDragonfiend) [inline code/controllers/master.dm:298]: To here.

**Review by EvilDragonfiend — APPROVED** [review]
I see nothing particularly wrong

File: code/controllers/master.dm (hunk 1 above)
Comment (Tsar-Salat) [reply code/controllers/master.dm:235]: Erroneous Deletion. I made sure to bring over the extra logging added a few months ago and bacon's sleep() improvements. Looks like I missed one though.

Comment (Tsar-Salat) [reply code/controllers/master.dm:235]: Readded in [0a29982](https://github.com/BeeStation/BeeStation-Hornet/pull/10933/commits/0a29982d23a9bf4b3abd963a8a11081bbde92e41)

File: code/controllers/master.dm (hunk 2 above)
Comment (Tsar-Salat) [reply code/controllers/master.dm:298]: Readded in [0a29982](https://github.com/BeeStation/BeeStation-Hornet/pull/10933/commits/0a29982d23a9bf4b3abd963a8a11081bbde92e41)

**Comment by EvilDragonfiend** [comment]
Putting Approval Label.
Even though the removal of configuration would make no difference, by the policy, Crossed should know this.

**Comment by EvilDragonfiend** [comment]
unknown weird game interactions were there, but I am not sure if it's related to this subsystem. I will do this as a single testmerge someday.

**Comment by EvilDragonfiend** [comment]
xenoarch TM failed. I focus on this again.

**Comment by EvilDragonfiend** [comment]
it appears it breaks readied player count

**Comment by Tsar-Salat** [comment]
> it appears it breaks readied player count

@EvilDragonfiend Fixed as of df32002

image

**Review by PowerfulBacon — APPROVED** [review]
Seems fine, TM'd for a good amount of time

**Pull request merged**